src/Adapter/Store/QueryHandler/GetStoreHandler.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Store\QueryHandler;
+
+use Store;
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\StoreException;
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\StoreNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Store\Query\GetStore;
+use PrestaShop\PrestaShop\Core\Domain\Store\QueryHandler\GetStoreHandlerInterface;
+
+use PrestaShopException;
+
+class GetStoreHandler implements GetStoreHandlerInterface
+{
+	/**
+	 * {@inheritdoc}
+	 *
+	 * @throws StoreException
+	 */
+	public function handle(GetStore $query): Store
+	{
+		try {
+			$store = new Store($query->getStoreId()->getValue());
+
+			if (0 >= $store->id) {
+				throw new StoreNotFoundException(sprintf('Store object with id %s was not found', var_export($query->getStoreId()->getValue(), true)));
+			}
+		} catch (PrestaShopException $e) {
+			throw new StoreException(sprintf('An unexpected error occurred when retrieving store with id %s', var_export($query->getStoreId()->getValue(), true)), 0, $e);
```

Comment:
```suggestion
			throw new StoreException(sprintf('An unexpected error occurred when retrieving store with id %d', $query->getStoreId()->getValue()), 0, $e);
```

`%d` permits to use integer (double) in `sprinf` sentence.